### ISSUE TITLE
Need to install en_core_web_sm to build

### ISSUE THREAD

**Issue by JohnnyUrosevic** [skekre98/NBA-Search#46]
**Describe the bug**
After following the build guide in the README I got the following error
```
OSError: [E050] Can't find model 'en_core_web_sm'. It doesn't seem to be a shortcut link, a Python package or a valid path to a data directory.
```
**To Reproduce**
1. Clone the repo
2. Install the requirements
3. `python main.py run`

**Expected behavior**
I would expect after installing all the pip modules it would be able to run right away

**Additional context**
I solved this by running `python3 -m spacy download en_core_web_sm`. I don't think there's anyway to add this to requirements.txt. Maybe we should create a setup script which installs that along with all the pip dependencies?

**Comment by skekre98** [comment]
I am actually unable to reproduce this issue. Can you check the logs when deploying the application. The language package should be installed when the application gets deployed:
https://github.com/skekre98/NBA-Search/blob/c79f0baef4748439778c6565cd907b81e820c743/app.py#L10

Perhaps the issue is persisting because of difference in python version?

**Comment by JohnnyUrosevic** [comment]
Here was the log before I ran the command manually
```
➜  NBA-Search git:(master) python main.py run
Removing previous training instance
No previous instance detected
Traceback (most recent call last):
  File "app.py", line 2, in <module>
    from modules.analysis import isNBA
  File "/home/goose/Code/NBA-Search/modules/analysis.py", line 9, in <module>
    nlp = spacy.load("en_core_web_sm")
  File "/home/goose/.local/lib/python3.8/site-packages/spacy/__init__.py", line 30, in load
    return util.load_model(name, **overrides)
  File "/home/goose/.local/lib/python3.8/site-packages/spacy/util.py", line 175, in load_model
    raise IOError(Errors.E050.format(name=name))
OSError: [E050] Can't find model 'en_core_web_sm'. It doesn't seem to be a shortcut link, a Python package or a valid path to a data directory.
```

Here is the log after
```
➜  NBA-Search git:(master) python main.py run                                        
Removing previous training instance
No previous instance detected
/home/goose/.local/lib/python3.8/site-packages/sklearn/externals/joblib/__init__.py:15: FutureWarning: sklearn.externals.joblib is deprecated in 0.21 and will be removed in 0.23. Please import this functionality directly from joblib, which can be installed with: pip install joblib. If this warning is raised when loading pickled models, you may need to re-serialize those models with scikit-learn 0.21+.
  warnings.warn(msg, category=FutureWarning)
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: en_core_web_sm==2.3.1 from https://github.com/explosion/spacy-models/releases/download/en_core_web_sm-2.3.1/en_core_web_sm-2.3.1.tar.gz#egg=en_core_web_sm==2.3.1 in /home/goose/.local/lib/python3.8/site-packages (2.3.1)
Requirement already satisfied: spacy<2.4.0,>=2.3.0 in /home/goose/.local/lib/python3.8/site-packages (from en_core_web_sm==2.3.1) (2.3.2)
Requirement already satisfied: cymem<2.1.0,>=2.0.2 in /home/goose/.local/lib/python3.8/site-packages (from spacy<2.4.0,>=2.3.0->en_core_web_sm==2.3.1) (2.0.3)
Requirement already satisfied: thinc==7.4.1 in /home/goose/.local/lib/python3.8/site-packages (from spacy<2.4.0,>=2.3.0->en_core_web_sm==2.3.1) (7.4.1)
Requirement already satisfied: requests<3.0.0,>=2.13.0 in /usr/lib/python3.8/site-packages (from spacy<2.4.0,>=2.3.0->en_core_web_sm==2.3.1) (2.24.0)
Requirement already satisfied: srsly<1.1.0,>=1.0.2 in /home/goose/.local/lib/python3.8/site-packages (from spacy<2.4.0,>=2.3.0->en_core_web_sm==2.3.1) (1.0.2)
Requirement already satisfied: catalogue<1.1.0,>=0.0.7 in /home/goose/.local/lib/python3.8/site-packages (from spacy<2.4.0,>=2.3.0->en_core_web_sm==2.3.1) (1.0.0)
Requirement already satisfied: preshed<3.1.0,>=3.0.2 in /home/goose/.local/lib/python3.8/site-packages (from spacy<2.4.0,>=2.3.0->en_core_web_sm==2.3.1) (3.0.2)
Requirement already satisfied: tqdm<5.0.0,>=4.38.0 in /home/goose/.local/lib/python3.8/site-packages (from spacy<2.4.0,>=2.3.0->en_core_web_sm==2.3.1) (4.50.2)
Requirement already satisfied: setuptools in /usr/lib/python3.8/site-packages (from spacy<2.4.0,>=2.3.0->en_core_web_sm==2.3.1) (50.3.0)
Requirement already satisfied: blis<0.5.0,>=0.4.0 in /home/goose/.local/lib/python3.8/site-packages (from spacy<2.4.0,>=2.3.0->en_core_web_sm==2.3.1) (0.4.1)
Requirement already satisfied: plac<1.2.0,>=0.9.6 in /home/goose/.local/lib/python3.8/site-packages (from spacy<2.4.0,>=2.3.0->en_core_web_sm==2.3.1) (1.1.3)
Requirement already satisfied: numpy>=1.15.0 in /home/goose/.local/lib/python3.8/site-packages (from spacy<2.4.0,>=2.3.0->en_core_web_sm==2.3.1) (1.18.5)
Requirement already satisfied: murmurhash<1.1.0,>=0.28.0 in /home/goose/.local/lib/python3.8/site-packages (from spacy<2.4.0,>=2.3.0->en_core_web_sm==2.3.1) (1.0.2)
Requirement already satisfied: wasabi<1.1.0,>=0.4.0 in /home/goose/.local/lib/python3.8/site-packages (from spacy<2.4.0,>=2.3.0->en_core_web_sm==2.3.1) (0.8.0)
Requirement already satisfied: chardet>=3.0.2 in /usr/lib/python3.8/site-packages (from requests<3.0.0,>=2.13.0->spacy<2.4.0,>=2.3.0->en_core_web_sm==2.3.1) (3.0.4)
Requirement already satisfied: idna>=2.5 in /usr/lib/python3.8/site-packages (from requests<3.0.0,>=2.13.0->spacy<2.4.0,>=2.3.0->en_core_web_sm==2.3.1) (2.10)
Requirement already satisfied: urllib3>=1.21.1 in /usr/lib/python3.8/site-packages (from requests<3.0.0,>=2.13.0->spacy<2.4.0,>=2.3.0->en_core_web_sm==2.3.1) (1.25.10)
✔ Download and installation successful
You can now load the model via spacy.load('en_core_web_sm')
 * Serving Flask app "app" (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)
127.0.0.1 - - [09/Oct/2020 13:36:49] "GET / HTTP/1.1" 200 -
127.0.0.1 - - [09/Oct/2020 13:36:49] "GET /static/img/banner_bg.png HTTP/1.1" 200 -
127.0.0.1 - - [09/Oct/2020 13:36:54] "GET /chat HTTP/1.1" 200 -
127.0.0.1 - - [09/Oct/2020 13:36:54] "GET /static/js/chatbox.js HTTP/1.1" 200 -
127.0.0.1 - - [09/Oct/2020 13:37:15] "POST /bot-msg HTTP/1.1" 200 -
127.0.0.1 - - [09/Oct/2020 13:37:36] "POST /bot-msg HTTP/1.1" 200 -
127.0.0.1 - - [09/Oct/2020 13:37:49] "POST /bot-msg HTTP/1.1" 200 -
127.0.0.1 - - [09/Oct/2020 13:38:04] "POST /bot-msg HTTP/1.1" 200 -
127.0.0.1 - - [09/Oct/2020 13:38:10] "GET /predictions HTTP/1.1" 200 -
127.0.0.1 - - [09/Oct/2020 13:38:11] "GET /favicon.ico HTTP/1.1" 404 -
```

**Comment by skekre98** [comment]
Oh this actually makes a lot of sense. I think a minor fix would be to move the install command above the module imports, but I like your idea of creating a setup script! This will likely be required when actually going to host the application anyways. I'll open an issue for it 👍. We can close this issue when it is completed.

**Comment by skekre98** [comment]
This issue will be addressed by #47

**Comment by skekre98** [comment]
Fixed with #52
